src/frontend/running.rs

```diff
@@ -352,6 +387,22 @@ impl RunningView {
                         {
                             self.farmer_state.initial_reward_address_balance -= decreased_by;
                         }
+
+                        // In case of balance increase or decrease, update the last reward timestamp
```

Comment:
Why would you care about decrease here? Also we already have decrease handled above, we just need to put this in `else` there if I understand correctly. And better as `Instant` rather thant `Option<u64>`, where `u64` sometimes seems to mean milliseconds and sometimes seconds in the code.